### PR TITLE
DashVtk examples

### PR DESCRIPTION
**Pull request by RohitRathore1**
Started to work on DashVtk:
1. Currently added about DashVtk and what we will cover
2. Working on Intro to 3D Visualization
3. Set up the all paths and default view

### CODE REVIEW

File: DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    dcc_markdown("
+    In VTK, we have 3 main types of objects that are key for understanding its visualization principals. First we have the *View* which is just a container for any *Representation* of DataSource that you want to see.
+
+    ## View
+    The view is a 3D View that can do Geometry rendering for meshes or Volume rendering for 3D images. The view can be configured to act as a 2D one when using parallel projection and preventing rotation when interacting with it. The *View* component can be configured with the following set of properties.
+
+    ```
+    dash_vtk.View(
+      id='vtk-view',
```
Comment (mohdibntarek) [inline DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl]: This is not Julia syntax. Please make sure this is runnable Julia code.

File: DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        action: 'Roll',
+        alt: true,
+        shift: true,
+      }
+    ]
+    ```
+
+    A mouse event can be identified with the following set of properties:
+    - *button*: 1/2/3 # Which button should be down
+    - *shift*: True/False # Is the `Shift` key down
```
Comment (mohdibntarek) [inline DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl]: True/False is not Julia syntax, I think it should be true/false

**Comment by mohdibntarek** [comment]
There seems to be a conflict with master. Please resolve it.

**Comment by RohitRathore1** [comment]
> There seems to be a conflict with master. Please resolve it.

Ok, Sure.

File: DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl (hunk 1 above)
Comment (RohitRathore1) [reply DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl]: Yes, I am looking into it and will keep in mind further. Thanks for the review.

**Comment by RohitRathore1** [comment]
Started to work on DashVtk:
1. Currently added about DashVtk and what we will cover
2. Working on Intro to 3D Visualization
3. Set up all paths and default view

Update:
Now, we have an example from Advanced Demos called Synthetic Volume Rendering which is related to medical and fixed the issues from previous reviews.

File: DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      triggerResetCamera=0,           # Timestamp meant to trigger a reset camera when different
+      # clickInfo,                    # Read-only property to retrieve picked representation id and picking information
+      # hoverInfo                     # Read-only property to retrieve picked representation id and picking information
+    )
+    ```
+
+    For the *interactorSettings* we expect a list of mouse event type linked to an action. The example below is what is used as default:
+    ```
+    interactorSettings=[
+      {
```
Comment (mohdibntarek) [inline DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl]: This should be a Julia Dict

**Comment by RohitRathore1** [comment]
Can we go for the implementation of [utils.py](https://github.com/plotly/dash-vtk/blob/master/dash_vtk/utils/vtk.py) in Julia because we have defined some functions which are required for the conversion of examples and the defined functions are: `to_mesh_state`, `to_volume_state`, `present_as_options`.?

**Comment by RohitRathore1** [comment]
Started to work on DashVtk:
1. Currently added about DashVtk and what we will cover
2. Working on Intro to 3D Visualization
3. Set up all paths and default view

Update:
Now, we have an example from Advanced Demos called Synthetic Volume Rendering which is related to medical and fixed the issues from previous reviews. 

Update:
In Structure of Dataset now we have converted examples of:
a) vtk_imagedata
b) vtk_polydata

**Comment by RohitRathore1** [comment]
Started to work on DashVtk:

Currently added about DashVtk and what we will cover
Working on Intro to 3D Visualization
Set up all paths and default view
Update:
Now, we have an example from Advanced Demos called Synthetic Volume Rendering which is related to medical and fixed the issues from previous reviews.

Update:
In Structure of Dataset now we have converted examples of:
a) vtk_imagedata
b) vtk_polydata

Update:
Now, we have a complete section of
1. Structure Dataset
2. Representation Components

File: DashDocs/src/app/default_page.jl [hunk 4]
```diff
@@ -88,9 +88,9 @@ function default_page()
                     "The Dash DAQ library contains a set of higher-level components like boolean switch, color picker, gauge and more."
                 ),
                 Chapter(
-                    "Dash DataTable",
```
Comment (mohdibntarek) [inline DashDocs/src/app/default_page.jl]: don't remove this

File: DashDocs/src/DashDocs.jl [hunk 5]
```diff
@@ -15,7 +15,7 @@ include("chapters/dash_core_components/index.jl")
 include("chapters/dash_html_components/index.jl")
 include("chapters/dash_daq/index.jl")
 include("chapters/dash_bio/index.jl")
-include("chapters/dash_data_table/index.jl");
```
Comment (mohdibntarek) [inline DashDocs/src/DashDocs.jl:18]: don't remove this, just add both

File: DashDocs/src/chapters/dash_vtk/RepresentationComponents/index.jl [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+        
+        ## vtk_pointcloudrepresentation
+        
+        The `vtk_pointcloudrepresentation` is just a helper using the following structure to streamline rendering a point cloud dataset. The code snippet below is not complete but it should provide you with some understanding of the kind of simplification that is happening under the hood.
+
+        ```
+        function vtk_pointcloudrepresentation(; kwargs...) 
+                 return vtk_geometryrepresentation(
+                     id=kwargs.get("id"),
+                     colorMapPreset=kwargs.get("colorMapPreset"),
```
Comment (mohdibntarek) [inline DashDocs/src/chapters/dash_vtk/RepresentationComponents/index.jl:263]: Using kwargs.get like that is not idiomatic in Julia. You probably want to do `kwargs[:colorMapPreset]`

**Review by mohdibntarek — APPROVED** [review]

File: DashDocs/src/chapters/dash_vtk/RepresentationComponents/index.jl (hunk 6 above)
Comment (mohdibntarek) [reply DashDocs/src/chapters/dash_vtk/RepresentationComponents/index.jl:263]: Please address this in a separate PR.

**Pull request merged**